### PR TITLE
tests: create_points.py unit tests

### LINKED ISSUE #49: Write unit tests for create_points
Now that we've got an MVP end-to-end pipeline and are starting to make things more sophisticated, we should implement unit tests. 

---

Some of our tests will involve testing the CLIs we've set up, and [this Typer documentation](https://typer.tiangolo.com/tutorial/testing/) will be relevant.

### PR DESCRIPTION
**Pull request by alissawang**
Unit tests for `create_points.py` with the creation of an `assets` directory to hold test data.

Not the most familiar with geodata so let me know if there are edge cases I missed!

Closes #49

### CODE REVIEW

File: tests/test_create_points.py [hunk 1]
```diff
@@ -14,3 +29,64 @@ def test_help():
...
+    new_coords = interpolate_along_line(test_line, test_dist)
+    assert isinstance(new_coords, MultiPoint)
+    assert len(new_coords.geoms) == int(test_line.length / test_dist)
+
+
+@pytest.mark.parametrize("invalid_value", [None, geometry.Point((1, 1))])
+def test_create_points_error(invalid_value):
+    test_df = gpd.read_file("tests/assets/test_gdf.shp").head()
+    test_df.loc[0, "geometry"] = invalid_value
+    with pytest.raises(Exception):
```
Comment (dragonejt) [inline tests/test_create_points.py:60]: If you know what exception will be raised by an invalid value (ValueError?), put that here instead of the generic `Exception` so that we still error on unexpected errors

**Review by dragonejt — CHANGES_REQUESTED** [review]
Overall well done, but a few changes could be made, mostly nitpicks

**Comment by dragonejt** [comment]
Also, changing your commit message to follow[ Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) is preferred

File: tests/assets/test_gdf.shp
[the remark this answers is not in the document]
Comment (alissawang) [reply tests/assets/test_gdf.shp:1]: Where would be the best place for this?

File: tests/test_create_points.py [hunk 2]
```diff
@@ -14,3 +29,64 @@ def test_help():
...
+
+
+def test_filter_by_highway_type_error():
+    bad_df = gpd.GeoDataFrame(columns=["not_highway"])
+    with pytest.raises(Exception):
+        filter_by_highway_type(bad_df, [])
+
+
+def test_filter_by_highway_type():
+    highway_vals = np.random.choice(DEFAULT_HIGHWAY_VALUES_TO_KEEP, size=10).tolist()
```
[the remark this answers is not in the document]
Comment (alissawang) [reply tests/test_create_points.py:41]: Not super familiar with `pytest-randomly`, what would that be for in this use case?

**Comment by alissawang** [comment]
Sorry for how long this is taking, I just started a new job and moved but things should settle down soon and I can give more time to this project!

File: tests/assets/test_gdf.shp
[the remark this answers is not in the document]
Comment (dragonejt) [reply tests/assets/test_gdf.shp:1]: Maybe wherever you're accessing the file in `test_create_points.py`? or maybe README

File: tests/test_create_points.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (dragonejt) [reply tests/test_create_points.py:40]: It executes the tests in a random order, and it also resets the random seed for `np.random.choice`. If you want to add it, you can just install it and you don't have to make any code changes I think

**Review by dragonejt — APPROVED** [review]

**Comment by dragonejt** [comment]
> Sorry for how long this is taking, I just started a new job and moved but things should settle down soon and I can give more time to this project!

No worries about the time commitment, we're all doing our best in the off time we have lol

File: tests/test_create_points.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (jayqi) [reply tests/test_create_points.py:40]: Opened an issue: https://github.com/AmericanRedCross/street-view-green-view/issues/80

File: tests/assets/test_gdf.shp
[the remark this answers is not in the document]
Comment (danbjoseph) [reply tests/assets/test_gdf.shp:1]: is this conversation resolved?

**Review by dragonejt — APPROVED** [review]

**Comment by danbjoseph** [comment]
Awesome, thank you @alissawang! Your contribution is greatly appreciated! And thank you to @dragonejt for reviewing!
